networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/BeaconChainMethods.java

```diff
@@ -145,15 +164,71 @@ public static BeaconChainMethods create(
           final RpcEncoding rpcEncoding) {
     final BeaconBlocksByRootMessageHandler beaconBlocksByRootHandler =
         new BeaconBlocksByRootMessageHandler(recentChainData);
-    return new Eth2RpcMethod<>(
-        asyncRunner,
-        BEACON_BLOCKS_BY_ROOT,
-        rpcEncoding,
-        BeaconBlocksByRootRequestMessage.SSZ_SCHEMA,
-        spec.getGenesisSchemaDefinitions().getSignedBeaconBlockSchema(),
-        true,
-        beaconBlocksByRootHandler,
-        peerLookup);
+    final SignedBeaconBlockSchema phase0BlockSchema =
+        spec.forMilestone(SpecMilestone.PHASE0).getSchemaDefinitions().getSignedBeaconBlockSchema();
+
+    final BeaconBlocksByRootRequestMessageSchema requestType =
+        BeaconBlocksByRootRequestMessage.SSZ_SCHEMA;
+    final boolean expectResponseToRequest = true;
+
+    final SingleProtocolEth2RpcMethod<BeaconBlocksByRootRequestMessage, SignedBeaconBlock>
+        v1Method =
+            new SingleProtocolEth2RpcMethod<>(
+                asyncRunner,
+                BEACON_BLOCKS_BY_ROOT,
+                1,
+                rpcEncoding,
+                requestType,
+                expectResponseToRequest,
+                encoding ->
+                    RpcResponseDecoder.createContextFreeDecoder(encoding, phase0BlockSchema),
+                beaconBlocksByRootHandler,
+                peerLookup);
+
+    if (spec.getForkSchedule().getSupportedMilestones().contains(SpecMilestone.ALTAIR)) {
```

Comment:
Well, `getSupportedMilestones` will always include Altair if we have Altair or any later fork scheduled.  And we actually want to run this in phase0 as long as we're supporting altair, so that we can smoothly transition across the fork.  If a peer isn't supporting v2, we'll negotiate the v1 version.  As we get closer to the fork, everyone should end up running v2 and we can at some point drop v1.

Reply:
Adding a shortcut `spec.isMilestoneSupported(...)` method

Reply:
I was thinking mostly of the case where we have a future Beetleguice fork and start a new chain from there.  I guess technically phase0 and Altair are supported but I thought they'd wind up getting dropped from the list because multiple forks would all start at slot 0.

Either way, I don't think we need to worry too much about this yet. The shortcut method is a good idea just for readability and will help find these cases if we need to adjust them in the future.

Reply:
Right - we have 2 styles of methods for inspecting milestones right now:
* Queries for "supported" milestones via methods like: `Spec.isMilestoneSupported`
* Queries for "active" milestones via `Spec.getEnabledMilestones` (should probably be renamed `getActiveMilestones` - it delegates to `ForkSchedule.getActiveMilestones`)

Maybe there's a clearer way to name these.  But the first style is meant to handle cases like you mention where we start up with Beetleguice enabled at slot 0, for example.  This helps with logic where we don't care if the milestone (altair, for example) is directly active, just that it is "implicitly" active.  I'd say altair is implicitly active in this case because the forks have a fixed ordering and not all logic is overwritten from one fork to the next.

The second style is to be used when we're trying to understand what active fork transitions we have to deal with.

Reply:
Yep that makes sense.  I think we'll get used to the naming.